protocol/adapter/datasync_whisper.go

```diff
@@ -96,7 +96,7 @@ func (w *DataSyncWhisperAdapter) decodePayload(message *whisper.ReceivedMessage)
 }
 
 func (w *DataSyncWhisperAdapter) decodeMessages(payload protobuf.Payload) []*protocol.Message {
-	messages := make([]*protocol.Message, len(payload.Messages))
```

Comment:
Oh I changed this but didn't change `messages = append(messages, &decoded)` to `messages[i] = &decoded`. And that's not really possible at the same time because some messages can be skipped.